terminus-ui/chip/src/chip-collection.component.ts

```diff
@@ -357,34 +346,43 @@ export class TsChipCollectionComponent implements AfterViewInit, OnInit, OnDestr
   @Output()
   public readonly tabUpdateFocus = new EventEmitter();
 
-  constructor(protected elementRef: ElementRef<HTMLElement>,
-              private changeDetectorRef: ChangeDetectorRef,
-              public zone: NgZone,
-              @Optional() private dir: Directionality,) {
+
+  constructor(
+    protected elementRef: ElementRef<HTMLElement>,
+    private changeDetectorRef: ChangeDetectorRef,
+    public zone: NgZone,
+  ) {
     zone.runOutsideAngular(() => {
       this._role = this.empty ? null : 'listbox';
     });
   }
 
-  public ngOnInit() {
+
+  /**
+   * Initialize the selection model
+   */
+  public ngOnInit(): void {
     this.selectionModel = new SelectionModel<TsChipComponent>(this.allowMultipleSelections, undefined, false);
   }
 
-  public ngAfterViewInit() {
+
+  /**
+   * Initialize the key manager and listen for chip changes
+   */
+  public ngAfterViewInit(): void {
     this.keyManager = new FocusKeyManager<TsChipComponent>(this.chips)
       .withWrap()
       .withVerticalOrientation()
-      .withHorizontalOrientation(this.dir ? this.dir.value : 'ltr');
+      .withHorizontalOrientation('ltr');
 
     this.keyManager.tabOut.pipe(untilComponentDestroyed(this)).subscribe(() => {
       this.tabUpdateFocus.emit();
     });
 
     // When the collection changes, re-subscribe
-    this.chips.changes.pipe(startWith(null), untilComponentDestroyed(this)).subscribe(() => {
+    this.chips.changes.pipe(startWith<void, null>(null), untilComponentDestroyed(this)).subscribe(() => {
```

Comment:
What does adding `<void, null>` buy you?

Reply:
It makes TypeScript happy. The version of `startWith` that doesn't have types passed in is deprecated while this version is not.

```typescript
startWith<T, D>(...array: Array<T | SchedulerLike>): OperatorFunction<T, T | D>
```